### PR TITLE
refactor: Removed protobuf encoding from zksync_types

### PR DESCRIPTION
**Pull request by pompon0**
## What ❔

Removed protobuf encoding from zksync_types.

## Why ❔

To make zksync_types have less dependencies.

### CODE REVIEW

File: core/lib/types/src/api/en.rs [hunk 1]
```diff
@@ -5,7 +5,12 @@ use zk_evm::ethereum_types::Address;
 use zksync_basic_types::{L1BatchNumber, MiniblockNumber, H256};
 use zksync_contracts::BaseSystemContractsHashes;
 
-use crate::{block::ConsensusBlockFields, ProtocolVersionId};
+use crate::ProtocolVersionId;
+
+/// Protobuf-encoded consensus-related L2 block (= miniblock) fields.
+/// See `zksync_dal::models::storage_sync::ConsensusBlockFields`.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ConsensusBlockFields(pub Vec<u8>);
```
[the remark this answers is not in the document]
Comment (pompon0) [reply core/lib/types/src/api/en.rs]: done

**Review by slowli — APPROVED** [review]
It still rubs me the wrong way that we need to define Protobuf types in the DAL crate (rather than the domain crate using DAL), but I see no sound way to move them to `zksync_core` while preserving the same level of type safety. E.g., it should be possible to define the consensus fields *trait* in the DAL crate that abstractly describes their conversions (to / from JSON and to Protobuf bytes) w/o tie-in to `zksync_protobuf`, and implement this trait on the `zksync_core` side. But this would look weird; and we'd probably cannot guarantee that a single trait implementation is used for each relevant DAL call.

File: core/lib/dal/src/models/storage_sync.rs [hunk 2]
```diff
@@ -24,45 +28,126 @@ pub struct StorageSyncBlock {
     pub consensus: Option<serde_json::Value>,
 }
 
+fn parse_h256(bytes: &[u8]) -> anyhow::Result<H256> {
+    Ok(<[u8; 32]>::try_from(bytes).context("invalid size")?.into())
+}
+
+fn parse_h160(bytes: &[u8]) -> anyhow::Result<H160> {
+    Ok(<[u8; 20]>::try_from(bytes).context("invalid size")?.into())
+}
```
[the remark this answers is not in the document]
Comment (pompon0) [reply core/lib/dal/src/models/storage_sync.rs:37]: yup, let's do it later. In fact I don't understand why these are not implemented by hash types natively.

File: core/lib/dal/src/models/storage_sync.rs [hunk 3]
```diff
@@ -1,8 +1,12 @@
+use anyhow::Context as _;
+use std::convert::TryFrom;
 use std::convert::TryInto;
```
[the remark this answers is not in the document]
Comment (pompon0) [reply core/lib/dal/src/models/storage_sync.rs]: done

**Review by popzxc — DISMISSED** [review]
I feel that it's still a bit odd that `zksync_dal` is currently a "source" of protobuf dependency, but I certainly like it more than the previous version. 
Obviously we can iterate if the current solution will turn out to be problematic in some sense, so I vote for merging as-is.

**Review by ml-ia — APPROVED** [review]

**Review by slowli — APPROVED** [review]

**Pull request merged**